server/conn.go

```diff
@@ -1047,6 +1048,9 @@ func (cc *clientConn) handleQuery(ctx context.Context, sql string) (err error) {
 		metrics.ExecuteErrorCounter.WithLabelValues(metrics.ExecuteErrorToLabel(err)).Inc()
 		return errors.Trace(err)
 	}
+	cc.mu.Lock()
```

Comment:
maybe we should do a check after obtain lock?....

think about this situation

Thread1: handle kill command  hold lock and doing for-loop close executor logic in `killConn`
At the same time Thread2 coming: and wait lock at this line...

After Thread1 done,  Thread2 got lock  and then continue execute resultSet event if connection be closed ?

Reply:
You are right.
Now after getting this lock, thread2 will check the status and if it find it has been killed by another connection, it will close these executors and return an error.

Reply:
PTAL~